ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound/State.hs

```diff
@@ -241,6 +242,7 @@ tickTimedTxs now st@SharedTxState{ timedTxs
                                  , bufferedTxs } =
     let (expiredTxs, timedTxs') = Map.split now timedTxs
         expiredTxs'             =  -- Map.split doesn't include the `now` entry in any map
+                                   -- TODO: there must be a better way.
```

Comment:
Short of getting the maintainers from Map to change their ways I cant think of one

Reply:
There's [`splitLookup`](https://hackage.haskell.org/package/containers-0.8/docs/Data-Map-Lazy.html#v:splitLookup) which reduces two traversals into one; subsequent `insert` is cheap, because it just inserts minimum element into the map.

Reply:
Fixed: https://github.com/IntersectMBO/ouroboros-network/pull/4887/commits/2a581d73851fa7d45c7f2dae4014c65ebefb7a61